README.md

```diff
@@ -46,7 +46,7 @@ let dtf = NeoFormatter::try_new(
 )
 .expect("locale should be present in compiled data");
 
-let date = DateTime::try_new_iso_datetime(2020, 9, 12, 12, 35, 0).expect("datetime should be valid");
+let date = DateTime::try_new_iso(2020, 9, 12, 12, 35, 0).expect("datetime should be valid");
```

Comment:
Out of scope for this PR, but can we make the calendar types be date factories instead? 

```rust
ISO.try_new_date(2020, 9, 12)
```

This way we'll only have a single `try_new_date` method on the calendar trait, not one name per calendar.

Reply:
It's nice to not have to import the calendar name. You can just import `Date` and call the fn.

Reply:
yeah, that's the goal of this.

Reply:
This definitely doesn't work with all the `_with_calendar` methods you're adding.

> You can just import Date and call the fn.

You *just* have to parse this list to figure out which fn you need. And of course this list doesn't use the same identifiers as calendar names
<img width="374" alt="image" src="https://github.com/user-attachments/assets/cac96ed3-105f-458c-a8a4-7e9401f697bd">

Reply:
> This definitely doesn't work with all the `_with_calendar` methods you're adding.
> 
> >

We can also add compiled data variants of those functions without `_with_calendar`, that's why they're disambiguated in the first place. I could add that now.

Reply:
That's blowing up that list even more, no?

Reply:
Yes.

Reply:
I see these as being convenience constructors. They shouldn't, and I think they don't, export functionality that you can't get by using `try_new_from_codes` which is the generic constructor. Right?

I guess I'm not opposed to `Date::<Gregorian>::try_new(...)` where you implement a `try_new` fn for all the calendars, all with the same name and disambiguated by the type param. We can leave `try_new_iso` as a special case.